apis/workloads/v1alpha1/instanceset_types.go

```diff
@@ -75,6 +75,66 @@ type SchedulingPolicy struct {
 	TopologySpreadConstraints []corev1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
 }
 
+// InstanceSetUpdateStrategyType is a string enumeration type that enumerates
+// all possible update strategies for the InstanceSet controller.
+// +enum
+type InstanceSetUpdateStrategyType string
+
+const (
+	// RollingUpdateInstanceSetStrategyType indicates that update will be
+	// applied to all Pods in the InstanceSet with respect to the InstanceSet
+	// ordering constraints. When a scale operation is performed with this
+	// strategy, new Pods will be created from the specification version indicated
+	// by the InstanceSet's updateRevision.
+	RollingUpdateInstanceSetStrategyType InstanceSetUpdateStrategyType = "RollingUpdate"
+	// OnDeleteInstanceSetStrategyType triggers the legacy behavior. Version
+	// tracking and ordered rolling restarts are disabled. Pods are recreated
+	// when they are manually deleted. When a scale
+	// operation is performed with this strategy, specification version indicated
+	// by the InstanceSet's currentRevision.
+	OnDeleteInstanceSetStrategyType InstanceSetUpdateStrategyType = "OnDelete"
```

Comment:
Do we still need "OnDelete"? @free6om

I believe it's unnecessary, as databases generally prefer in-place updates.

Reply:
No need anymore, can be removed.